sql/core/src/main/scala/org/apache/spark/sql/DataFrameStatFunctions.scala

```diff
@@ -536,6 +536,11 @@ final class DataFrameStatFunctions private[sql](df: DataFrame) {
    * @since 2.0.0
    */
   def bloomFilter(col: Column, expectedNumItems: Long, fpp: Double): BloomFilter = {
+    if (fpp <= 0D || fpp >= 1D) {
```

Comment:
Perhaps in a follow-up, we could try moving the fpp check to the `BloomFilter#optimalNumOfBits(long, double)` method?

Reply:
yeah, on second thought, it's worth a separate PR to unify the check of `fpp`.
let me remove the fpp check for now.

Reply:
OK